### ISSUE TITLE
Data source for aws_customer_gateway

### ISSUE THREAD

**Issue by henrydobson** [terraform-providers/terraform-provider-aws#5007]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

aws_customer_gateway data source.

Customer gateways are general unique. Creation may only occur once but when it's required to create additional VPN connections for VPCs the only way to reference the cgw id is by using an output, which may be in a different module or remote state once you start using workspaces. It would be better to have a data source for readability, accuracy, avoiding complex conditionals and lookups being used to extract the id from state or copping out and setting a variable.

### New or Affected Resource(s)

* data.aws_customer_gateway

### Potential Terraform Configuration

```hcl
data "aws_customer_gateway" "selected" {
  filter {
    name = "tag:Name"
    values = ["my-cgw"]
  }
}
```

### References

* #0000

**Comment by ripta** [comment]
I'd love to take a stab at this.

**Comment by bflad** [comment]
A new `aws_customer_gateway` data source has been merged and will release with version 2.15.0 of the Terraform AWS Provider, likely later today. Thanks, @ripta! 🚀
